### PR TITLE
Color dialog

### PR DESCRIPTION
**Pull request by nallath**
CURA-8938

### CODE REVIEW

File: resources/qml/Preferences/Materials/MaterialsView.qml [hunk 1]
```diff
@@ -222,7 +222,22 @@ Item
                     id: colorDialog
                     title: catalog.i18nc("@title", "Material color picker")
                     color: properties.color_code
-                    onAccepted: base.setMetaDataEntry("color_code", properties.color_code, color)
+//                    swatchColors: ["#2161AF", "#57AFB2", "#F7B32D", "#E33D4A", "#C088AD"]
```
Comment (nallath) [inline resources/qml/Preferences/Materials/MaterialsView.qml:225]:
```suggestion
```

File: resources/qml/Preferences/Materials/MaterialsView.qml [hunk 2]
```diff
@@ -222,7 +222,22 @@ Item
...
+                    onAccepted: {
+                        base.setMetaDataEntry("color_code", properties.color_code, color);
+                        console.log("color_code");
+
+                        const timer = Qt.createQmlObject("import QtQuick 2.0; Timer {}", base);
+                        timer.interval = 10000;
+                        timer.repeat = true;
+                        timer.triggered.connect(function () {
+                            console.log("updating colors");
```
Comment (nallath) [inline resources/qml/Preferences/Materials/MaterialsView.qml:234]:
```suggestion
```

File: resources/qml/Preferences/Materials/MaterialsView.qml [hunk 3]
```diff
@@ -222,7 +222,22 @@ Item
...
+                            colorDialog.swatchColors = ["#2161AF", "#57AFB2", "#F7B32D", "#E33D4A", "#C088AD"];
+                        })
+
+                        console.log("starting timers");
```
Comment (nallath) [inline resources/qml/Preferences/Materials/MaterialsView.qml:238]:
```suggestion
```

File: resources/qml/Preferences/Materials/MaterialsView.qml [hunk 4]
```diff
@@ -222,7 +222,22 @@ Item
                     id: colorDialog
                     title: catalog.i18nc("@title", "Material color picker")
                     color: properties.color_code
-                    onAccepted: base.setMetaDataEntry("color_code", properties.color_code, color)
+//                    swatchColors: ["#2161AF", "#57AFB2", "#F7B32D", "#E33D4A", "#C088AD"]
+                    onAccepted: {
+                        base.setMetaDataEntry("color_code", properties.color_code, color);
+                        console.log("color_code");
```
Comment (nallath) [inline resources/qml/Preferences/Materials/MaterialsView.qml:228]:
```suggestion
```

File: resources/qml/ColorDialog.qml [hunk 5]
```diff
@@ -17,61 +17,133 @@ UM.Dialog
...
+                        color: UM.Theme.getColor("checkbox")
+                    }
+
+                    MouseArea
+                    {
+                        anchors.fill: parent
+                        onClicked:
+                        {
+                            base.color = swatchColor;
+                        }
```
Comment (nallath) [inline resources/qml/ColorDialog.qml:96]:
```suggestion
                        onClicked: base.color = swatchColor
```

File: resources/qml/ColorDialog.qml [hunk 6]
```diff
@@ -17,61 +17,133 @@ UM.Dialog
...
-                text = `#${text}`;
+                model: ListModel
+                {
+                    id: swatchColorsModel
+                }
+
+                delegate: Rectangle
+                {
+                    color: swatchColor
+                    width: 24
```
Comment (nallath) [inline resources/qml/ColorDialog.qml:78]: Not really a big issue, but shouldn't these be themed? Right now the sizes will be different depending on the DPI

File: resources/qml/Preferences/Materials/MaterialsView.qml (hunk 4 above)
Comment (casperlamboo) [reply resources/qml/Preferences/Materials/MaterialsView.qml:228]: Oops forgot to remove some debugging code

File: resources/qml/ColorDialog.qml (hunk 6 above)
Comment (casperlamboo) [reply resources/qml/ColorDialog.qml:78]: Would you introduce these _new_ sizes? They are the same as a "wide margin" but it doesn't feel right to pick those

Comment (nallath) [reply resources/qml/ColorDialog.qml:78]: Maybe button_icon? Or extruder icon?

File: resources/qml/Preferences/Materials/MaterialsView.qml (hunk 1 above)
Comment (casperlamboo) [reply resources/qml/Preferences/Materials/MaterialsView.qml]: https://github.com/Ultimaker/Cura/pull/11546/commits/c8be7832c8b6379166ab2f92232b595e41cf82a4

File: resources/qml/Preferences/Materials/MaterialsView.qml (hunk 2 above)
Comment (casperlamboo) [reply resources/qml/Preferences/Materials/MaterialsView.qml]: https://github.com/Ultimaker/Cura/pull/11546/commits/c8be7832c8b6379166ab2f92232b595e41cf82a4

File: resources/qml/Preferences/Materials/MaterialsView.qml (hunk 3 above)
Comment (casperlamboo) [reply resources/qml/Preferences/Materials/MaterialsView.qml]: https://github.com/Ultimaker/Cura/pull/11546/commits/c8be7832c8b6379166ab2f92232b595e41cf82a4

File: resources/qml/ColorDialog.qml (hunk 5 above)
Comment (casperlamboo) [reply resources/qml/ColorDialog.qml:96]: https://github.com/Ultimaker/Cura/pull/11546/commits/c8be7832c8b6379166ab2f92232b595e41cf82a4

**Pull request merged**